New try to get these changes committed (Translations - one file deleted and replaced)

mARTin-B78 opened the pull request.

gitforkgit commented on Projects/Foundations/Schemas/Docs-Tutorials/D/Developers/Developers-Intro/developers-intro-001-tutorial-intro-for-developers.json:
```diff
@@ -10,6 +10,11 @@
+                "language": "DE",
+                "text": "Dieses Tutorial führt Sie durch die technischen Aspekte von Superalgos, so dass Sie die Vorteile der vorhandenen Infrastruktur so effizient wie möglich nutzen können.",
+                "updated": 1641729113684
+                },
+                {
```
It would be great if you could align both lines more to the left. You can orient on the part down below how far to the left.(You can do it here on github by clicking on the file and then edit file)

gitforkgit commented on Projects/Foundations/Schemas/Docs-Tutorials/D/Developers/Developers-Intro/developers-intro-002-tutorial-topic-system-architecture.json:
```diff
@@ -10,6 +10,11 @@
+                "language": "DE",
+                "text": "Superalgos ist im Wesentlichen eine Node.js-Anwendung + eine Web-App.",
+                "updated": 1641729195697
+                },
+                {
```
Same here.

gitforkgit commented on Projects/Foundations/Schemas/Docs-Tutorials/D/Developers/Developers-Intro/developers-intro-004-tutorial-topic-client.json:
```diff
@@ -10,6 +10,11 @@
+                "language": "DE",
+                "text": "Ist eine Sammlung von Servern, die auf einem einzigen Node.js-Prozess laufen.",
+                "updated": 1641729279193
+                },
+                {
```
Here the same. I think this is so at many files.

gitforkgit commented on Projects/Foundations/Schemas/Docs-Tutorials/D/Developers/Developers-Intro/developers-intro-005-tutorial-step-web-server.json:
```diff
@@ -10,6 +10,11 @@
+                "language": "DE",
+                "text": "Der Webserver hat zwei Hauptziele: Er liefert alle JS-Dateien, die die Webanwendung bilden, und alle Datendateien, die in den Diagrammen gezeichnet werden sollen, an den Browser, der die Benutzeroberfläche ausführt.",
+                "updated": 1641729528371
+                },
+                {
```
Here the same.

gitforkgit commented on Projects/Foundations/Schemas/Docs-Tutorials/D/Developers/Developers-Intro/developers-intro-006-tutorial-step-web-sockets-server.json:
```diff
@@ -10,6 +10,11 @@
+                "language": "DE",
+                "text": "Dies ist eine einfache Implementierung eines Web-Sockets-Servers. Er ermöglicht es der Benutzeroberfläche oder anderen Diensten, über WS mit dem Eventserver zu kommunizieren.",
+                "updated": 1641729694113
+                },
+                {
```
Same here.

gitforkgit commented on Projects/Foundations/Schemas/Docs-Tutorials/D/Developers/Developers-Intro/developers-intro-007-tutorial-step-events-server.json:
```diff
@@ -10,6 +10,11 @@
+                "language": "DE",
+                "text": "Der Eventserver stellt das Kommunikationsmodell zwischen Prozessen und der Benutzeroberfläche bereit. Er erlaubt im Wesentlichen jeder Entität innerhalb eines Prozesses, einen systemweiten Event-Handler zu erstellen und jeder anderen Entität innerhalb eines Prozesses (oder der Benutzeroberfläche) zu erlauben, Events bei diesem Event-Handler zu abonnieren oder zu emittiere ...
+                "updated": 1641729769800
+                },
+                {
```
Same here.

gitforkgit commented on Projects/Foundations/Schemas/Docs-Tutorials/D/Developers/Developers-Intro/developers-intro-008-tutorial-step-task-manager-server.json:
```diff
@@ -10,6 +10,11 @@
+                "language": "DE",
+                "text": "Der Task Manager Server ist ein einfacher Server, der die Ausführung von Task Server Node.js Prozessen verwaltet. Er behält den Überblick über alle Instanzen, die zu einem bestimmten Zeitpunkt laufen, und ermöglicht es der Benutzeroberfläche, zu entscheiden, wann weitere Instanzen ausgeführt oder eine der laufenden Instanzen gestoppt werden soll.",
+                "updated": 1641729859188
+                },
+                {
```
Same here.

gitforkgit commented on Projects/Foundations/Schemas/Docs-Tutorials/D/Developers/Developers-Intro/developers-intro-009-tutorial-topic-task-server.json:
```diff
@@ -10,6 +10,11 @@
+                "language": "DE",
+                "text": "Der Task-Server ist ein Node.js-Fork-Prozess des Clients, der die Definition des Tasks, der ausgeführt werden soll, von der Benutzeroberfläche erhält.",
+                "updated": 1641729950857
+                },
+                {
```
Same here.

gitforkgit commented on Projects/Foundations/Schemas/Docs-Tutorials/D/Developers/Developers-Intro/developers-intro-010-tutorial-topic-frontend.json:
```diff
@@ -10,6 +10,11 @@
+                "language": "DE",
+                "text": "Das Frontend ist im Wesentlichen eine Javascript-Webapplikation, die im Browser läuft.",
+                "updated": 1641730035452
+                },
+                {
```
Same here

mARTin-B78 replied:
looks like someone fixed it already

mARTin-B78 replied:
looks like someone fixed it already

gitforkgit commented on Projects/Foundations/Schemas/Docs-Tutorials/D/Developers/Developers-Intro/developers-intro-011-tutorial-topic-canvas-app.json:
```diff
@@ -10,7 +10,11 @@
+                "language": "DE",
+                "text": "Der größte Teil der Aktion läuft über eine HTML 5 Canvas-Animation. Der Rest ist reines HTML + CSS, sonst nichts.",
+                "updated": 1641730136254
+        },
+        {                "language": "RU",
```
The text was inserted a bit wrong as well as the brackets. No big Problem.

gitforkgit commented on Projects/Foundations/Schemas/Docs-Tutorials/D/Developers/Developers-Intro/developers-intro-011-tutorial-topic-canvas-app.json:
```diff
@@ -59,7 +71,11 @@
             "text": "Design Space: Where definitions are maintained.",
             "translations": [
                 {
-                    "language": "RU",
+                    "language": "DE",
+                    "text": "Design Space: Hier werden die Definitionen verwaltet.",
+                    "updated": 1641730167618
+        },
+        {                    "language": "RU",
                     "text": "Design Space: Где сохраняются определения.",
                     "updated": 1640930671870
                 },
```
Here you accidentally overwrote the russian translation.

gitforkgit commented on Projects/Foundations/Schemas/Docs-Tutorials/D/Developers/Developers-Intro/developers-intro-011-tutorial-topic-canvas-app.json:
```diff
@@ -171,7 +213,11 @@
             "text": "Top Space: Currently, the top bar with logo.",
             "translations": [
                 {
-                    "language": "RU",
+                    "language": "DE",
+                    "text": "Top Space: Zurzeit die obere Leiste mit dem Logo.",
+                    "updated": 1641730241268
+        },
+        {                    "language": "RU",
                     "text": "Top Space: В настоящее время верхняя панель с логотипом.",
```
Same for the whole File.

gitforkgit commented on Projects/Foundations/Schemas/Docs-Tutorials/D/Developers/Developers-Intro/developers-intro-012-tutorial-step-canvas-animation.json:
```diff
@@ -10,7 +10,11 @@
         "translations": [
             {
-                "language": "RU",
+                "language": "DE",
+                "text": "Die Animation führt für jeden Raum eine physics()- und eine draw()-Funktion aus, so dass jeder Raum seine eigene Physik-Engine bzw. seine eigenen Regeln hat und sich selbst als oberste Ebene auf der Benutzeroberfläche darstellen kann.",
+                "updated": 1641730326103
+        },
+        {                "language": "RU",
```
Same in this File.

gitforkgit commented on Projects/Foundations/Schemas/Docs-Tutorials/D/Developers/Developers-Intro/developers-intro-013-tutorial-step-charting-space.json:
```diff
@@ -10,7 +10,11 @@
         "text": "This is where the charts are displayed. It allows users to navigate through multi-dimensional datasets.",
         "translations": [
             {
-                "language": "RU",
+                "language": "DE",
+                "text": "Hier werden die Diagramme angezeigt. Er ermöglicht es den Benutzern, durch mehrdimensionale Datensätze zu navigieren.",
+                "updated": 1641735239256
+        },
+        {                "language": "RU",
```
Same problem here.

gitforkgit commented on Projects/Foundations/Schemas/Docs-Tutorials/D/Developers/Developers-Intro/developers-intro-014-tutorial-step-design-space.json:
```diff
@@ -10,7 +10,11 @@
         "translations": [
             {
-                "language": "RU",
+                "language": "DE",
+                "text": "Verwaltet das Kernkonzept der Knoten, die, wenn sie miteinander verbunden werden, umfangreiche Definitionen bilden können, die von den Benutzern leicht erstellt und gepflegt werden können.",
+                "updated": 1641735783104
+        },
+        {                "language": "RU",
                 "text": "Управляет основной схемой узлов, которые, будучи присоединенными друг к другу, могут образовывать обширные определения, легко создаваемые и поддерживаемые пользователями.",
                 "updated": 1640931112373
             }
```
Same for this File.

gitforkgit commented on Projects/Foundations/Schemas/Docs-Tutorials/D/Developers/Developers-Intro/developers-intro-015-tutorial-step-floating-space.json:
```diff
@@ -10,7 +10,11 @@
         "text": "This is like the low-level part of the Design Space. It features a physics engine that governs the physical interactions of floating objects.",
         "translations": [
             {
-                "language": "RU",
+                "language": "DE",
+                "text": "Dies ist sozusagen der Low-Level-Teil des Design Space. Er verfügt über eine Physik-Engine, die die physikalischen Interaktionen der schwebenden Objekte steuert.",
+                "updated": 1641735953807
+        },
```
Same for this File.

gitforkgit commented on Projects/Foundations/Schemas/Docs-Tutorials/T/Testing/Testing-a-Trading-Idea-The-Trigger-Stage/testing-a-trading-idea-the-trigger-stage-035-tutorial-step-lets-see-what-weve-got.json:
```diff
@@ -10,11 +10,17 @@
+                "language": "DE",
+                "text": "Wir werden nun die Simulation durchlaufen und sehen, ob die von uns erstellte Definition die erwarteten Ergebnisse bringt!",
+                "updated": 1641763341271
+                },
+                {
```
Brackets inserted wrong in this file.

gitforkgit commented on Projects/Foundations/Schemas/Docs-Tutorials/T/Testing/Testing-a-Trading-Idea-The-Trigger-Stage/testing-a-trading-idea-the-trigger-stage-037-tutorial-step-cant-see-whats-expected.json:
```diff
@@ -10,6 +10,11 @@
+                "language": "DE",
+                "text": "Wenn Sie nicht sehen können, was ich in dem Video gezeigt habe, haben Sie wahrscheinlich eine oder mehrere meiner vorherigen Anweisungen übersprungen!",
+                "updated": 1641763445408
+                },
+                {
```
Brackets inserted wrong in this file.

mARTin-B78 reviewed (commented):
Can this be undone?

mARTin-B78 reviewed (commented):
I still do not see the issue here

gitforkgit commented on Projects/Foundations/Schemas/Docs-Tutorials/D/Developers/Developers-Intro/developers-intro-020-tutorial-topic-files-component.json:
```diff
@@ -33,7 +41,11 @@
                 {
-                    "language": "RU",
+                    "language": "DE",
+                    "text": "Market Files: Für Zeitrahmen von 1 Stunde und mehr ist es praktisch, eine einzige Datei mit allen Daten des gesamten Marktes zu haben.",
+                    "updated": 1641736760116
+        },
+        {                    "language": "RU",
```
Nearly all files look great now. The only ones left are the some files where: "language": "RU",   is too far right and the brackets are to far left.

gitforkgit reviewed (changes requested):
Nearly all files look great now. There are some left. Could you please go trough the files and make the last fixes where language:"RU" ist to far right and the brackets to far left.
And the file you removed at the beginning: Projects/Foundations/Schemas/Docs-Tutorials/A/Aggregate/Aggregate-data-from-different-Exchanges-in-a-Chart-/aggregate-data-from-different-exchanges-in-a-chart-001-combine-exchanges-in-a-chart-tutorial.json

is missing at the end. Maybe you accidentally canceled that the original file get`s added at the end.

harrellbm commented:
@mARTin-B78 @gitforkgit It's so so so close now!  Make sure to restore the original this file Projects/Foundations/Schemas/Docs-Tutorials/A/Aggregate/Aggregate-data-from-different-Exchanges-in-a-Chart-/aggregate-data-from-different-exchanges-in-a-chart-001-combine-exchanges-in-a-chart-tutorial.json

And then everything else is good to go 👍🏼

mARTin-B78 commented:
Trying to restore the file: Superalgos\Projects\Foundations\Schemas\Docs-Tutorials\A\Aggregate\Aggregate-data-from-different-Exchanges-in-a-Chart-\ aggregate-data-from-different-exchanges-in-a-chart-001-combine-exchanges-in-a-chart-tutorial.json

updated my respo at github. (Fetch upstream)
ran the "app.update command" to make sure my local files are up to date
Made sure the file above exists on my local installation of Superalgos.
added the changes in Notepad++ (Spanish / German Translations)
ran the app.contribute command to restore the file.
got the "Your contributions were uploaded to your Superalgos Fork and a Pull Request was submitted to the main Superalgos repository." Message..

What else can I do to solve it?

gitforkgit commented on Projects/Foundations/Schemas/Docs-Tutorials/D/Developers/Developers-Intro/developers-intro-020-tutorial-topic-files-component.json:
```diff
@@ -77,11 +103,15 @@
-                    "language": "RU",
+                    "language": "DE",
+                    "text": "Tägliche Dateien werden von einem Dateicursor verwaltet, der dynamisch die Dateien eines Tages und seiner Nachfolgetage lädt.",
+                    "updated": 1641736822077
+        },
+        {                    "language": "RU",
                     "text": "Daily Files управляются File Cursor, который динамически загружает файлы одного дня и соседних с ним дней.",
                     "updated": 1640932087275
                 }
```
One of the parts I would afterwards change.

mARTin-B78 replied:
have lined up the brackets

gitforkgit reviewed (commented):
Hey @mARTin-B78 @harrellbm  i looked trough the files again and nearly all files look awesome. Only about 4-5 files have the brackets to far left.... @harrellbm If you could review the files and can approve they are correct (by their function) I would offer me to afterwards "correct" the visual issues so this pull request can get merged. If I understood it right the file that was deleted at the beginning was added directly as the second file in this Pull request. This File is not similar to the one in the master branch because @mARTin-B78  added the german translation but that shouldnt be a problem asit replaces the old file with a new one including another translation. @harrellbm If there is no problem with this proposal please let me know.

mARTin-B78 commented on Projects/Foundations/Schemas/Docs-Tutorials/T/Testing/Testing-a-Trading-Idea-The-Trigger-Stage/testing-a-trading-idea-the-trigger-stage-035-tutorial-step-lets-see-what-weve-got.json:
```diff
@@ -9,16 +9,21 @@
+                }, 
+                {
+                    "language": "TR",
+                    "text": "Simülasyonu gözden geçireceğiz ve ortaya çıkan tanımın beklenen sonuçlara sahip olup olmadığına bakacağız!",
+                    "updated": 1641655373874
+                },
+                {
+                    "language": "RU",
+                    "text": "Мы собираемся провести симуляцию и посмотреть, дает ли определение, которое мы придумали, ожидаемые результаты!",
+                    "updated": 1642655108942
+                }
```
I checked that file and they all line up... the old once and the new translations...
Maybe that is just displayed a bit to the right in github. 
like the line numbers in green are a bit to the right and the one in red are a bit to the left.

gitforkgit commented:
@harrellbm can you have a look at it?

harrellbm commented:
@gitforkgit @mARTin-B78 everything looks good except I'm still seeing aggregate-data-from-different-exchanges-in-a-chart-001-combine-exchanges-in-a-chart-tutorial.json getting deleted. What we could do at this point is just merge and then grab the file and add it back in an new PR since it seems other things aren't working
